### PR TITLE
 add field droppable feature 

### PR DESCRIPTION
**Pull request by SNsir**

### CODE REVIEW

File: galite-core/src/main/kotlin/org/kopi/galite/form/dsl/FormBlock.kt [hunk 1]
```diff
@@ -192,10 +203,10 @@ open class FormBlock(var buffer: Int, var visible: Int, ident: String, val title
   /**
    * Initializes a field.
    */
-  inline fun <reified T: Comparable<T>> initField(domain: Domain<T>,
-                                                  init: FormField<T>.() -> Unit,
-                                                  access: Int,
-                                                  position: FormPosition? = null): FormField<T> {
+  inline fun <reified T : Comparable<T>> initField(domain: Domain<T>,
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/form/dsl/FormBlock.kt:206]: The part to add the field with droppable is implemented insed the initField method with this part in line 1868 in FormParser.java  , but you will need to find alternatives for VKstringtype , VKImageType and reportTrouble() function  , afterwards the implementation will look like this : 

```javascript
    inline fun <reified T : Comparable<T>> initField(domain: Domain<T>,
                                                       init: FormField<T>.() -> Unit,
                                                       access: Int,
                                                       position: FormPosition? = null): FormField<T> {
        domain.kClass = T::class
        val field = FormField(this, domain, blockFields.size, access, position)
        field.init()
        field.setInfo()
    
        if (dropList == null) {
          blockFields.add(field)
        } else {
          if (type.getDef() !is VStringType
                  && type.getDef() !is VKImageType) {
            reportTrouble(PositionedError(sourceRef,
                    FormMessages.UNSUPPORTED_DROP_FIELD_TYPE,
                    name))
          } else {
            val flavor: String? = this.addDropList(dropList!!, field)
            if (flavor == null) {
              blockFields.add(field)
            } else {
              reportTrouble(PositionedError(sourceRef,
                      FormMessages.DUPLICATE_DROP_FLAVOR,
                      flavor,
                      this.dropListMap[flavor]))
            }
          }
        }
        return field
      }
```

File: galite-core/src/main/kotlin/org/kopi/galite/form/dsl/FormBlock.kt [hunk 2]
```diff
@@ -66,9 +66,9 @@ open class FormBlock(var buffer: Int, var visible: Int, ident: String, val title
   var blockTables: MutableList<FormBlockTable> = mutableListOf()
   var indices: MutableList<FormBlockIndex> = mutableListOf()
   var access: IntArray = IntArray(3) { VConstants.ACS_MUSTFILL }
+  var dropListMap = HashMap<String, String>()
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/form/dsl/FormBlock.kt:69]: Also Add property :   var dropList : MutableList<String>? = null

File: galite-core/src/main/kotlin/org/kopi/galite/form/VBlock.kt [hunk 3]
```diff
@@ -2999,7 +2999,7 @@ abstract class VBlock(var form: VForm) : VConstants, DBContextHandler, ActionHan
   var maxColumnPos = 0
   var displayedFields = 0
   private var isFilterVisible = false
-  protected var dropListMap = HashMap<Any?, Any?>()
+  protected var dropListMap = HashMap<String, String>()
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VBlock.kt:3002]: I dont think we need to add this to VBlock

File: galite-core/src/main/kotlin/org/kopi/galite/form/dsl/FormBlock.kt [hunk 4]
```diff
@@ -425,11 +436,14 @@ open class FormBlock(var buffer: Int, var visible: Int, ident: String, val title
...
         super.source = source ?: sourceFile
+        super.name = ident
         super.bufferSize = buffer
         super.pageNumber = this@FormBlock.pageNumber
         super.maxRowPos = this@FormBlock.maxRowPos
         super.maxColumnPos = this@FormBlock.maxColumnPos
-        super.name = ident
+        this@FormBlock.dropListMap.forEach{
+                  super.dropListMap.put(it.key,it.value)
+        }
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/form/dsl/FormBlock.kt:446]: We probably don't need this , we will recheck after implementing the other corrections

**Review by ZMarouani — CHANGES_REQUESTED** [review]

**Review by ZMarouani — APPROVED** [review]

**Review by hfazai — APPROVED** [review]

**Pull request merged**